pkg/crd/flatten.go

```diff
@@ -37,6 +37,15 @@ type ErrorRecorder interface {
 	AddError(error)
 }
 
+// WarningRecorder knows how to record errors. It wraps the part of
+// pkg/loader.Package that we need to record errors in places were it might not
+// make sense to have a loader.Package
```

Comment:
the second sentence here seems like it was copy-pasted verbatim, and doesn't make much sense any more.
I'd go with

```suggestion
// WarningRecorder knows how to record warnings, which are effectively
// non-fatal errors.  It wraps the part of pkg/loader.Package that we need to
// record warnings in places were it might not make sense to have a
// loader.Package.
```